src/Sulu/Component/Content/Types/Rlp/Strategy/RlpStrategy.php

```diff
@@ -184,61 +221,42 @@ private function adaptResourceLocators(ResourceSegmentBehavior $document, $userI
         $webspaceKey = $this->documentInspector->getWebspace($document);
         $languageCode = $this->documentInspector->getLocale($document);
 
-        foreach ($document->getChildren() as $childDocument) {
-            $childNode = $this->documentInspector->getNode($childDocument);
-
-            // determine structure
-            $templatePropertyName = $this->nodeHelper->getTranslatedPropertyName('template', $languageCode);
+        $node = $this->documentInspector->getNode($document);
+        $node->getSession()->save();
 
-            if (!$childNode->hasProperty($templatePropertyName)) {
+        foreach ($document->getChildren() as $childDocument) {
+            // skip documents without assigned resource segment
+            if (!$childDocument instanceof ResourceSegmentBehavior
+                || !$currentResourceLocator = $childDocument->getResourceSegment()) {
+                $this->adaptResourceLocators($childDocument, $userId);
                 continue;
             }
 
+            // build new resource segment based on parent changes
+            $parentUuid = $this->documentInspector->getUuid($document);
+            $childPart = $this->getChildPart($currentResourceLocator);
+            $newResourceLocator = $this->generate($childPart, $parentUuid, $webspaceKey, $languageCode);
+
+            // save new resource locator
+            $childNode = $this->documentInspector->getNode($childDocument);
+            $templatePropertyName = $this->nodeHelper->getTranslatedPropertyName('template', $languageCode);
             $template = $childNode->getPropertyValue($templatePropertyName);
             $structure = $this->structureManager->getStructure($template);
 
-            if (!$structure->hasTag('sulu.rlp')) {
-                continue;
-            }
-
-            // get rlp
-            try {
-                $rlp = $this->loadByContent($childDocument);
-            } catch (ResourceLocatorNotFoundException $ex) {
-                $childNode->getSession()->save();
-
-                $rlpPart = $childNode->getPropertyValue(
-                    $this->nodeHelper->getTranslatedPropertyName('title', $languageCode)
-                );
-                $parentRlp = $this->mapper->getParentPath(
-                    $childNode->getIdentifier(),
-                    $webspaceKey,
-                    $languageCode
-                );
-
-                // generate new resourcelocator
-                $rlp = $this->generate(
-                    $rlpPart,
-                    $parentRlp,
-                    $webspaceKey,
-                    $languageCode
-                );
-            }
-
-            // build new resource segment based on parent changes
-            $rlpParts = explode('/', $rlp);
-            $newRlp = $document->getResourceSegment() . '/' . $rlpParts[count($rlpParts) - 1];
```

Comment:
That's the important part missing now, also see https://github.com/sulu/sulu/pull/2706/files#r73861675